src/main/java/org/breedinginsight/utilities/Utilities.java

```diff
@@ -162,8 +162,21 @@ public static Object formatBrapiObjForDisplay(Object brapiInstance, Class brapiC
         return brapiInstance;
     }
 
+    /**
+     * \s*: Matches zero or more whitespace characters before the opening bracket.
+     * \[: Matches the opening square bracket [. The backslash is used to escape the special meaning of [ in regex.
+     * .*?: Matches any character (except newline) zero or more times, non-greedily.
+     * . matches any character except newline.
+     * * means "zero or more times".
+     * ? makes the matching non-greedy, so it stops at the first closing bracket.
+     * \]: Matches the closing square bracket ]. Again, the backslash is used to escape it.
+     * \s*: Matches zero or more whitespace characters after the closing bracket.
+     * @param original
+     * @param programKey
+     * @return
+     */
```

Comment:
Unit tests for this method covering these cases with specific assertions for the expected format of synonyms and pedigree strings aster stripping would be helpful when testing for regressions later. New edge cases found could be added along with tests for very large fields or complex pedigrees.